### PR TITLE
feat(TDP-3703): enable display modes in lists

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-3703

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1170/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1170/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1171/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1171/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1183/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1183/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1188/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1188/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1190/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1190/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1192/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1192/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1195/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1195/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1199/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1199/)

File: dataprep-webapp/src/app/components/home/dataset/home-dataset-container.spec.js [hunk 1]
```diff
@@ -55,4 +55,9 @@ describe('Home Dataset Container', () => {
 		//then
 		expect(element.find('#inputUpdateDataset').length).toBe(1);
 	});
+
+	if('should render display mode dropdown', () => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/home/dataset/home-dataset-container.spec.js]: `it` instead

File: dataprep-webapp/src/app/components/home/dataset/home-dataset-container.spec.js [hunk 2]
```diff
@@ -55,4 +55,9 @@ describe('Home Dataset Container', () => {
...
+		createElement();
+		expect(element.find('datasets-list-display-mode').length).toBe(1);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/home/dataset/home-dataset-container.spec.js]: `expect(element.find('.datasets-list-display-mode').length).toBe(1);` instead

File: dataprep-webapp/src/app/components/home/preparation/home-preparation-container.spec.js [hunk 3]
```diff
@@ -65,4 +65,9 @@ describe('Home Preparation Container', () => {
 		//then
 		expect(element.find('inventory-list').length).toBe(1);
 	});
+
+	if('should render display mode dropdown', () => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/home/preparation/home-preparation-container.spec.js]: copy/paste :)

File: dataprep-webapp/src/app/components/home/preparation/home-preparation-container.spec.js [hunk 4]
```diff
@@ -65,4 +65,9 @@ describe('Home Preparation Container', () => {
...
+		createElement();
+		expect(element.find('preparations-list-display-mode').length).toBe(1);
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/home/preparation/home-preparation-container.spec.js]: copy/paste :)

File: dataprep-webapp/src/app/components/widgets-containers/inventory-list/inventory-list-container.js [hunk 5]
```diff
@@ -32,6 +32,7 @@ const InventoryListContainer = {
 			list="$ctrl.listProps"
 			toolbar="$ctrl.toolbarProps"
 			watch-depth="reference"
+			virtualized="true"
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/inventory-list/inventory-list-container.js]: Should be removed

File: dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js [hunk 6]
```diff
@@ -142,6 +146,27 @@ describe('Inventory state service', () => {
...
+
+		it('should not persist the datasets display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
+			inventoryState.datasetsDisplayMode = 'table';
+			InventoryStateService.setDatasetsDisplayMode({ mode: 'table' });
+			expect(StorageService.setListsDisplayModes).not.toHaveBeenCalled();
+		}));
+
+		it('should not persist the datasets display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
+			inventoryState.datasetsDisplayMode = 'table';
+			InventoryStateService.setDatasetsDisplayMode({ mode: 'table' });
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js]: useless

File: dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js [hunk 7]
```diff
@@ -142,6 +146,27 @@ describe('Inventory state service', () => {
...
+			expect(inventoryState.datasetsDisplayMode).toBe('large');
+			expect(StorageService.setListsDisplayModes).toHaveBeenCalledWith({
+				datasets: 'large',
+				preparations: 'table',
+			});
+		}));
+
+		it('should not persist the datasets display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
+			inventoryState.datasetsDisplayMode = 'table';
+			InventoryStateService.setDatasetsDisplayMode({ mode: 'table' });
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js]: useless

File: dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js [hunk 8]
```diff
@@ -142,6 +146,27 @@ describe('Inventory state service', () => {
...
+			expect(StorageService.setListsDisplayModes).not.toHaveBeenCalled();
+		}));
+
+		it('should not persist the datasets display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js]: same test

File: dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js [hunk 9]
```diff
@@ -180,6 +205,27 @@ describe('Inventory state service', () => {
...
+			});
+		}));
+
+		it('should not persist the preparations display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
+			inventoryState.preparationsDisplayMode = 'table';
+			InventoryStateService.setPreparationsDisplayMode({ mode: 'table' });
+			expect(StorageService.setListsDisplayModes).not.toHaveBeenCalled();
+		}));
+
+		it('should not persist the preparations display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js]: same same

File: dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js [hunk 10]
```diff
@@ -180,6 +205,27 @@ describe('Inventory state service', () => {
...
+			InventoryStateService.setPreparationsDisplayMode({ mode: 'large' });
+			expect(inventoryState.preparationsDisplayMode).toBe('large');
+			expect(StorageService.setListsDisplayModes).toHaveBeenCalledWith({
+				preparations: 'large',
+				datasets: 'table',
+			});
+		}));
+
+		it('should not persist the preparations display mode if it is the same', inject((inventoryState, InventoryStateService, StorageService) => {
+			inventoryState.preparationsDisplayMode = 'table';
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/inventory/inventory-state-service.spec.js]: should be removed

File: dataprep-webapp/src/app/services/utils/storage/storage-service.spec.js [hunk 11]
```diff
@@ -382,8 +382,27 @@ describe('Storage service', () => {
...
+		it('should get lists display modes', inject(($window, StorageService) => {
+			expect(StorageService.getListsDisplayModes()).toEqual({});
+			$window.localStorage.setItem(LISTS_DISPLAY_MODES_KEY, JSON.stringify(mock));
+			const modes = StorageService.getListsDisplayModes();
+			expect(modes).toEqual(mock);
+		}));
+
+		it('should save lists display modes', inject(($window, StorageService) => {
+			StorageService.setListsDisplayModes(mock);
+			expect($window.localStorage.getItem(LISTS_DISPLAY_MODES_KEY)).toBe('{"datasets":"large","preparations":"table"}');
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/utils/storage/storage-service.spec.js]: JSON.stringify(mock) instead

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1203/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1203/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1205/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1205/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1230/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1230/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/2/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/2/)

**Pull request merged**